package/MDAnalysis/analysis/nuclinfo.py

```diff
@@ -763,3 +769,62 @@ def pseudo_dihe_baseflip(universe, bp1, bp2, i,
     pseudo = mdamath.dihedral(x[0] - x[1], x[1] - x[2], x[2] - x[3])
     pseudo = np.rad2deg(pseudo) % 360
     return pseudo
+
+
+def angle_between_base_planes(universe, b1, b2, seg1="SYSTEM", seg2="SYSTEM"):
+    """The angle between the planes for given two bases is computed.
+
+    See [Gabb1996]_ and [Jafilan2012]_ for background information.
+
+    .. Note:: This angle calculation will only work if using atom names as
+              documented by charmm force field parameters.
+
+    Parameters
+    ----------
+    universe : Universe
+        :class:`~MDAnalysis.core.universe.Universe` containing the
+        trajectory
+    b1 : int
+        resid of the first base
+    b2 : int
+        resid of the second base
+    segid1 : str (optional)
+        segid of b1
+    segid2 : str (optional)
+        segid of b2
+
+    Returns
+    -------
+    float
+          angle in degrees
+
+
+    .. versionadded:: 2.1.0
+    """
+    batms = {'CYT': ['N1', 'N3', 'C5'],
+             'THY': ['N1', 'N3', 'C5'],
+             'URA': ['N1', 'N3', 'C5'],
+             'ADE': ['N9', 'N1', 'N7'],
+             'GUA': ['N9', 'N1', 'N7']}
+    # select residues
+    bf1 = universe.select_atoms(
+        "(segid {0!s} and resid {1!s})".format(seg1, b1))
+    bf2 = universe.select_atoms(
+        "(segid {0!s} and resid {1!s})".format(seg2, b2))
+    # get residue names
+    resn1 = bf1.atoms.resnames[0]
+    resn2 = bf2.atoms.resnames[0]
+    # extract positions for specific atoms of each base
+    c11 = bf1.select_atoms("name {0!s}".format(batms[resn1][0])).positions[0]
```

Comment:
This will throw a KeyError if the residue is not part of batms and an IndexError if the names are wrong (since you're doing a positions call on a zero-width AtomGroup).

In both cases this will be quite incomprehensible to users if it fails. Could this be reformatted to add some try/excepts to make it clear to users as to why things fail (if they do)?

Reply:
In this case we can only have the 5 residues that make up the DNA and RNA systems. We already inform the users that this analysis module is based on CHARMM specific names for nucleic acid residues and atom names. Again, the user just needs to check that the segid and resid are correct.

Reply:
This is putting a lot of onus on users to follow our docs and fully understand everything we mean. Personally, given that wrapping in a try/except here has nearly no cost and will immensely clarify failures, I fail to see how this is not worth it.

Again happy to defer to the opinions of @orbeckst as the original approving reviewer here.

Reply:
I agree with @IAlibay here — even though it's the user's job to put in the right selection, telling them if they don't is a better approach than just failing with unintelligible stack traces (because then _we_ have to explain on the mailing list what the error means). 

The other functions don't have these additional checks but your function could do better than the old ones. (And this is not an API change but usability improvement.)

Reply:
Please take a look. Thanks!